aea/helpers/dialogue/base.py

```diff
@@ -406,6 +406,17 @@ def update(self, message: Message) -> bool:
         ):
             self._update_self_initiated_dialogue_label_on_second_message(message)
 
+        counterparty = None
```

Comment:
Should be `counterparty = None  # type: Optional[str]`

Reply:
Done